### ISSUE TITLE
[csolution] Add command dedicated for solution configuration

### ISSUE THREAD

**Issue by brondani** [Open-CMSIS-Pack/devtools#727]
Currently each context configuration is done implicitly when running the `convert` command where at a high level the following steps are executed:

- Read yml files
- Check files against schema (disable: `--no-check-schema`)
- Parse input yml nodes
- Load packs for selected contexts (control: `--load [latest|all|required])
- For each selected context: (disable: `--context ...`)
  - Filter pack selection
  - Get device/board selection
  - Get user groups/files
  - Find selected components
  - Process GPDSC if needed
  - Update configuration files (RTE) (disable: `--no-update-rte`)
  - Print dependency validation results
  - Generate CPRJ
  - Generate cbuild.yml
- Generate cbuild-idx.yml

For checking solution/context correctness and update configuration files **without** generating output files (CPRJ, cbuild.yml, cbuild-idx.yml) I would propose the intoduction of a dedicated command, for example `configure <csolution.yml> [-c <context>]`.

**Comment by jkrech** [comment]
Update configuration files includes the generation of the RTE_Components.h and pre-include files and is controlled by `--update-rte` ?

**Comment by brondani** [comment]
> Update configuration files includes the generation of the RTE_Components.h and pre-include files and is controlled by `--update-rte` ?

Yes it comprehends RTE_Components.h and pre-includes and in `csolution` this step is disabled by the option `--no-update-rte`.
`cbuildgen` has the mirrored behaviour and the option `--update-rte`.

**Comment by jkrech** [comment]
@ReinhardKeil this request is for only updating the RTE without generating any *.cbuild.yml / *.cprj files. This is not yet available in the latest csolution version 1.5.0.

**Comment by mcgordonite** [comment]
The `configure` command would be useful for the Software Components view in the csolution extension and similar flows. Here, we want to update the RTE directory to reflect any added components, but we don't want to build the whole solution or generated CPRJ files.

**Comment by brondani** [comment]
Further discussions:
- name the command `pre-configure`?
- list added and "outdated" configuration files, for the user to manually configure selected components?
@edriouk is there already an RteModel API to get such information?

**Comment by edriouk** [comment]
@brondani,  
We have moved the config file PLM handling out of RTE in the sense that 
- @base file indicates the original file version used for the last update.
- @update file indicates that file is outdated

There is no "added files" API. 
It can be easily added, and would be beneficial because would allow to copy files in one shot

**Comment by brondani** [comment]
The `update-rte` command has been implemented as [agreed](https://github.com/Open-CMSIS-Pack/devtools/issues/727#issuecomment-1550017785).

In verbose mode an info message is printed with the config files, "base" and "update" versions, for example:
```
info csolution: config files for each component:
  ARM::Device:Startup&RteTest Startup@2.0.3:
    - D:/work/devtools-wip/build/tools/projmgr/test/data/TestSolution/TestProject1/RTE/Device/RteTest_ARMCM0/ARMCM0_ac6.sct (base@1.0.0)
    - D:/work/devtools-wip/build/tools/projmgr/test/data/TestSolution/TestProject1/RTE/Device/RteTest_ARMCM0/startup_ARMCM0.c (base@1.0.0) (update@2.0.3)
    - D:/work/devtools-wip/build/tools/projmgr/test/data/TestSolution/TestProject1/RTE/Device/RteTest_ARMCM0/system_ARMCM0.c (base@1.0.0)
```

**Comment by jkrech** [comment]
@ReinhardKeil this command needs to be added to the documentation.
Purpose: Update `RTE` component configuration files and generated header files (e.g. RTE_Components.h) for specified contexts without generating cprj files.
When running in verbose mode all component configuration files are listed including their versions.
a) providing the user a list of active configuration files
b) providing the user information about configuration file "mismatches"
Please review to what extend this fulfills https://github.com/Open-CMSIS-Pack/devtools/issues/922

**Comment by jkrech** [comment]
Available starting CMSIS-Toolbox 2.0.0-dev1
